lib/index.js

```diff
@@ -196,13 +197,31 @@ Socket.prototype._stopPollReceive = function () {
  */
 
 Socket.prototype.bind = function (addr) {
-    return this._protect(nn.Bind(this.binding, addr));
+  var eid = nn.Bind( this.binding, addr );
+  if(eid < 0){
+    return this.emit('error', new Error(nn.Err() +': '+this.type+' bind@' + addr));
+  }
+  this.how[addr] = [eid,'bind'];
+  return this.how; //socket's active endpoints
 }
 
 Socket.prototype.connect = function (addr) {
-    return this._protect(nn.Connect(this.binding, addr));
+  var eid = nn.Connect( this.binding, addr )
+  if(eid < 0){
+    return this.emit('error', new Error(nn.Err() +': '+this.type+' connect@' + addr));
+  }
+  this.how[addr] = [eid,'connect'];
+  return this.how; //socket's active endpoints
```

Comment:
I'm good with that because it adds some transparency if you're just checking `console.log` on the socket quickly for bound and/or connected addresses.

but I'd rather call it:

```js
this.bound[addr] = eid;
```